routes/api.js

```diff
@@ -199,8 +216,10 @@ var killKernel = function(kernelId) {
         method: 'DELETE',
         headers: headers
     }, function(err, response, body) {
-        debug('PROXY: kill kernel response: ' +
-            response.statusCode + ' ' + response.statusMessage);
+        if (response) {
```

Comment:
Second to logging it. Not sure a retry makes sense. From the kernel gateway side, I believe the only way a DELETE for a kernel can happen is if the kernel doesn't exist.